src/handlers/handle-scheduled-checkin.ts

```diff
@@ -119,3 +123,23 @@ function adjustMessageVisibilityTimeout(input: AdjustMessageVisibilityTimeoutInp
 
   return client.send(command);
 }
+
+async function getHeadersWithRetry<HeaderType>(func: () => Promise<HeaderType>,
+  headerName: string, maxRetries: int)
+{
+  const waitMs = util.promisify(setTimeout);
+  while(true)
+  {
+    try{	
+	  const headers = await func();
+	  return headers;
+    }
+    catch(error){
+	  if(maxRetries <= 0)
+	    throw error;
```

Comment:
Rather than throwing an error, just return here and update the caller to check if the result is truthy and deal with changing the control flow if it isn't.